src/__tests__/unit/ui/link/test.tsx

```diff
@@ -202,18 +203,9 @@ describe('<Link />', () => {
       const component = wrapper.find('Link');
 
       component.simulate('click', baseClickEvent);
-
-      expect(HistoryMock.push).toHaveBeenCalledTimes(1);
-      expect(HistoryMock.push).toHaveBeenCalledWith(createPath(location));
-    });
-
-    it('should navigate correctly when `to` is a location object', () => {
```

Comment:
No it's not. Was never typed as supported either. And I think supporting `to` accepting 2 different object shapes (route and location) is bad. So I'm in favour of killing location support on `to` (it's still supported in `href` tho)

Reply:
Wouldn't it make more sense to keep the location support on `to` rather than `href`? HTML anchors support a string value, so maintaining that on the API and then using `to` for more custom stuff makes more sense to me

Reply:
Killed that entirely as was never officially supported